src/filters.hpp

```diff
@@ -23,9 +27,9 @@ class Filters : public Nan::ObjectWrap {
 
     static Nan::Persistent<v8::FunctionTemplate>& constructor();
 
-    void add_filter(filter_key_type&& key, filter_value_type&& filter, zoom_type minzoom, zoom_type maxzoom) {
+    void add_filter(filter_key_type&& key, filter_value_type&& filter, filter_properties properties, zoom_type minzoom, zoom_type maxzoom) {
```

Comment:
This should be passed as `filter_properties && properties` which means the object is movable in memory. Then you will need to use `std::move` to pass the variable to this function. See https://github.com/mapbox/cpp/blob/master/glossary.md#heap-allocation for more details on `std::move`.

Reply:
👍 got it(pass the memory into the next target), changed in next submit.